dipy/segment/metricspeed.pyx

```diff
@@ -44,12 +44,12 @@ cdef class Metric(object):
         def __get__(Metric self):
             return bool(self.is_order_invariant)
 
-    cdef int c_are_compatible(Metric self, Shape shape1, Shape shape2) nogil:
+    cdef int c_are_compatible(Metric self, Shape shape1, Shape shape2) nogil except -1:
         """ Cython version of `Metric.are_compatible`. """
         with gil:
             return self.are_compatible(shape2tuple(shape1), shape2tuple(shape2))
 
-    cdef double c_dist(Metric self, Data2D features1, Data2D features2) nogil:
+    cdef double c_dist(Metric self, Data2D features1, Data2D features2) nogil except -1.0:
```

Comment:
BTW - ran a try_branch with this:
http://nipy.bic.berkeley.edu/builders/dipy-py2.7-osx-10.8/builds/336

:-)

On Mon, Mar 16, 2015 at 10:21 AM, Marc-Alexandre Côté <
notifications@github.com> wrote:

> In dipy/segment/metricspeed.pyx
> <https://github.com/nipy/dipy/pull/598#discussion_r26506797>:
>
> >          """ Cython version of `Metric.are_compatible`. """
> >          with gil:
> >              return self.are_compatible(shape2tuple(shape1), shape2tuple(shape2))
> >
> > -    cdef double c_dist(Metric self, Data2D features1, Data2D features2) nogil:
> > +    cdef double c_dist(Metric self, Data2D features1, Data2D features2) nogil except -1.0:
>
> Ok, just tested it, it can be a int. Do you want me to make the change
> everywhere?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/598/files#r26506797>.
>